java/integTest/build.gradle

```
@@ -155,7 +155,31 @@ tasks.withType(Test) {
     afterTest { desc, result ->
         logger.quiet "${desc.className}.${desc.name}: ${result.resultType} ${(result.getEndTime() - result.getStartTime())/1000}s"
     }
+
+    finalizedBy jacocoTestReport, jacocoTestCoverageVerification
+}
+
+// JaCoCo section (code coverage by integration tests)
+jacocoTestReport {
+    sourceDirectories.setFrom(files("${project.rootDir}/client/src/main/java"))
+    classDirectories.setFrom(files("$buildDir/classes"))
+    executionData.setFrom(
+        files("$buildDir/jacoco/test.exec"),
+        files("$buildDir/jacoco/pubsubTest.exec")
+    )
+}
+jacocoTestCoverageVerification {
+    violationRules {
+        rule {
+            element = 'CLASS'
+            excludes = [
+                'glide.models.protobuf.*',
+                'glide.com.google.'
```

```suggestion
                'glide.com.google.*'
```
and add netty please